Sources/ForageSDK/Foundation/Telemetry/ForageLogger.swift

```diff
@@ -0,0 +1,102 @@
+//
+//  File.swift
+//
+//
+//  Created by Danilo Joksimovic on 2023-07-26.
+//
+
+import Datadog
+import Foundation
+
+// TODO: move this?
+protocol ForageLogger {
+    func setCustomerID(_ customerID: String)
+    func setMerchantAccount(_ merchantAccount: String)
+    func initialize(_ environment: EnvironmentTarget)
+
+    func notice(_ message: String, attributes: [String: Encodable]?)
+    func info(_ message: String, attributes: [String: Encodable]?)
+    func warn(_ message: String, error: Error?, attributes: [String: Encodable]?)
+    func error(_ message: String, error: Error?, attributes: [String: Encodable]?)
+    func critical(_ message: String, error: Error?, attributes: [String: Encodable]?)
+}
+
+public class DatadogLogger: ForageLogger {
+    static let shared = DatadogLogger()
+    private static let DD_CLIENT_TOKEN: String = "pub1e4572ba0f5e53df108c333d5ec66c02"
+    private static var logger: Logger? = nil
+
+    private init() {}
+
+    internal func initialize(_ environment: EnvironmentTarget) {
+        if DatadogLogger.logger != nil {
+            return
+        }
+
+        // TODO: only initialize if not running locally?
```

Comment:
I think we want to emit logs while running locally. The only scenario I think we should avoid logging is when we are running unit tests.

Reply:
657bf39a9c0b05d39ce7e681271fb19e7bd8447c